specifyweb/specify/views.py

```diff
@@ -464,6 +464,52 @@ def resolve_record_merge_response(start_function, silent=True):
     ('agent', 'spauditlog'): ['createdbyagent_id', 'modifiedbyagent_id']
 }
 
+# TODO: Refactor this to always use query sets.
+def clean_fields_pre_delete(obj_instance):
+    if (not obj_instance.__class__.__name__.endswith('attachment')
+            or not hasattr(obj_instance, 'attachment')) :
+        return
+    # We delete this object anyways. So, don't care about
+    # the value we put in here. If an error, everything is rollbacked.
+    obj_instance.attachment.attachmentlocation = None
+
+ordering_tables = {
+    table_name.lower(): fields for table_name, fields in orderings.items()
+}
+
+def fix_orderings(base_model: Table, new_record_data):
+    for field_name, records in list(new_record_data.items()):
+        try:
+            relationship = base_model.get_relationship(field_name)
+        except FieldDoesNotExistError:
+            continue
+        ordering_fields = ordering_tables.get(relationship.relatedModelName.lower(), None)
+        if (ordering_fields is None or
+                # Can this ever happen?
+                not isinstance(records, list)):
+            continue
+        order_fields_data = set([tuple([record.get(ordering_field, None)
+                              for ordering_field in ordering_fields])
+                                for record in records])
```

Comment:
While probably not something we need to consider immediately, at least for Django we _do_ also include `rankid` in the list of ordering fields (if applicable) outside of the global `orderings` dictonary

https://github.com/specify/specify7/blob/e572df6fdcb365e5537f50b326c0ebc10035715c/specifyweb/specify/build_models.py#L56-L59

Reply:
Thanks for catching this!